### PR TITLE
RUM-5004 Record SwiftUI Images

### PR DESCRIPTION
**Pull request by maxep**
### What and why?

Record SwiftUI images.

Proposed implementation will record all images regardless of the privacy level. This should be tackled in complementary work on SwiftUI and fine grained masking.

### How?

Reflect `GraphicsImage` and build wireframe and associated resource.

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [x] Run unit tests for Session Replay

### CODE REVIEW

File: DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIHostingViewRecorder.swift [hunk 1]
```diff
@@ -186,6 +186,40 @@ internal struct UIHostingWireframesBuilder: NodeWireframesBuilder {
...
             )
+        case let .image(graphics):
+            return graphics.contents.map { contents in
+                builder.createImageWireframe(
+                    id: Int64(content.seed.value),
+                    resource: UIImageResource(
+                        image: UIImage(
+                            cgImage: contents.cgImage,
+                            scale: graphics.scale,
+                            orientation: {
```
Comment (mariedm) [inline DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIHostingViewRecorder.swift:197]: nit: we could extract the `graphics.orientation` switch statement into a separate function or computed property for cleaner code.

Probably as well on `maskColor` used below.

**Review by mariedm — APPROVED** [review]
LGMT! 🙌

Just a nit, but it's a personal preference so feel free to disregard.

File: DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIHostingViewRecorder.swift (hunk 1 above)
Comment (maxep) [reply DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIHostingViewRecorder.swift:197]: Yes, let me change that 👍

**Review by mariedm — APPROVED** [review]
Awesome, thanks! 🚀

**Review by mariedm — APPROVED** [review]

**Pull request merged**